website/cue/reference/components/sinks/splunk_hec.cue

```diff
@@ -0,0 +1,47 @@
+package metadata
+
+components: sinks: _splunk_hec: {
+	configuration: {
+		acknowledgements: {
+			common:      false
+			description: "The configuration for Splunk HEC indexer acknowledgement client behavior."
+			required:    false
+			type: object: {
+				options: {
+					query_interval: {
+						common:      false
+						description: "The amount of time to wait in between queries to the Splunk HEC indexer acknowledgement endpoint. Minimum of `1`."
+						required:    false
+						type: uint: {
+							default: 10
+							unit:    "seconds"
+						}
+					}
+					retry_limit: {
+						common:      false
+						description: "The maximum number of times an ack id will be queried for its status. Minimum of `1`."
+						required:    false
+						type: uint: {
+							default: 30
+							unit:    null
+						}
+					}
+				}
+			}
+		}
+	}
+	how_it_works: {
```

Comment:
I'd like to see us address how this works with Vector's internal acknowledgements feature. Does it tie in appropriately? For example, will ack's flow back up the pipeline to the source?

Reply:
Sure, I can flesh this out. It does tie in to the end-to-end acknowledgements feature and makes it more accurate when used with `splunk_hec` sinks. The current behavior of acking solely based on a `200 OK` response is not as robust since Splunk HEC can still lose the data after that response.

Reply:
I added this detail in 6938ae2, let me know if that makes sense.

Reply:
Looks good to me.